### PR TITLE
Feature/add jepsen tests

### PR DESCRIPTION
**Pull request by SerialVelocity**
Adds jepsen testing. Have fun!
For #541

### CODE REVIEW

File: atlasdb-ete-tests/src/main/java/com/palantir/atlasdb/cas/CasClient.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+import com.google.common.base.Optional;
+import com.palantir.atlasdb.config.AtlasDbConfig;
+import com.palantir.atlasdb.factory.TransactionManagers;
+import com.palantir.atlasdb.table.description.Schema;
+import com.palantir.atlasdb.transaction.api.TransactionManager;
+
+import io.dropwizard.jersey.setup.JerseyEnvironment;
+
+public class CasClient {
```
Comment (joelea) [inline atlasdb-ete-tests/src/main/java/com/palantir/atlasdb/cas/CasClient.java]: Avoid abbreviations :(

File: atlasdb-ete-tests/src/main/java/com/palantir/atlasdb/cas/CasValueAccessor.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+package com.palantir.atlasdb.cas;
+
+import java.util.Objects;
+
+import com.google.common.base.Optional;
+import com.palantir.atlasdb.cas.generated.CasSchemaTableFactory;
+import com.palantir.atlasdb.cas.generated.CasTable;
+import com.palantir.atlasdb.transaction.api.Transaction;
+
+public class CasValueAccessor {
```
Comment (joelea) [inline atlasdb-ete-tests/src/main/java/com/palantir/atlasdb/cas/CasValueAccessor.java]: PersistentValue

File: atlasdb-ete-tests/src/main/java/com/palantir/atlasdb/cas/CasValuePatch.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.cas;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
+
+public class CasValuePatch {
```
Comment (joelea) [inline atlasdb-ete-tests/src/main/java/com/palantir/atlasdb/cas/CasValuePatch.java]: CasValueUpdate

File: atlasdb-ete-tests/src/main/java/com/palantir/atlasdb/todo/TodoResource.java [hunk 4]
```diff
@@ -25,7 +25,7 @@
 import javax.ws.rs.core.MediaType;
 
 @Path("/todos")
-public interface AtlasTodos {
+public interface TodoResource {
```
Comment (joelea) [inline atlasdb-ete-tests/src/main/java/com/palantir/atlasdb/todo/TodoResource.java]: CheckAndSetResource?

File: atlasdb-ete-tests/src/main/java/com/palantir/atlasdb/todo/TodoSchema.java [hunk 5]
```diff
@@ -25,15 +25,15 @@
...
     private static final Schema INDEX_TEST_SCHEMA = generateSchema();
-    public static final String TODOS_TABLE = "todos";
+    public static final String TODO_TABLE = "todo";
     public static final String TEXT_COLUMN = "text";
 
     private static Schema generateSchema() {
         Schema schema = new Schema();
 
-        schema.addTableDefinition(TODOS_TABLE, new TableDefinition() {{
+        schema.addTableDefinition(TODO_TABLE, new TableDefinition() {{
```
Comment (joelea) [inline atlasdb-ete-tests/src/main/java/com/palantir/atlasdb/todo/TodoSchema.java]: Should this now be not a TODO thing?

File: atlasdb-jepsen-tests/build.gradle [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+task showComposeLogs(type: Exec) {
+    mustRunAfter composeUp
+
+    environment = project.dockerCompose.environment
+    commandLine project.dockerCompose.composeCommand('logs', '-f')
+}
+
+task test(dependsOn: ['composeUp', 'showComposeLogs']) {
+    finalizedBy 'composeDown'
+} << {
```
Comment (joelea) [inline atlasdb-jepsen-tests/build.gradle]: Simplify as per conversation.

**Comment by SerialVelocity** [comment]
All "Todo" stuff is still "Todo", this adds a new endpoint for "CheckAndSet" and refactors the server/configuration to have both endpoints.

**Comment by jboreiko** [comment]
Would be great to see an addition to the docs describing how to run the testing.

**Comment by gbrova** [comment]
@jboreiko I *think* `./gradlew :atlasdb-ete-tests:test` is all you need, but let me know if that didn't work for some reason.

**Pull request merged**